src/views/KlantDetailView.vue

```diff
@@ -56,7 +82,7 @@ const klantBsn = computed(() => klant.value?.bsn);
 watch(
   klantId,
   (id) => {
-    if (!id) {
+    if (!id || !route.path.includes("klanten")) {
```

Comment:
When navigating from the Klant detail page (there's a id param here) to the Zaak detail page (there's a **new** id param here), this `watch()` was hit, resulting in a `fetchKlant()` call with the **new** Zaak detail id. This was not intended functionality (as this would always result in a 400).

This path check fixes it, might be a better solution to it though, if you know one, let me know.

Reply:
Good catch! I think a bit neater might be to rename the route parameter to klantId. We need to do this in this file, and in  router/index.ts as well.

Reply:
Fixed